assert/seq_supported.go

```diff
@@ -0,0 +1,44 @@
+//go:build go1.23 || goexperiment.rangefunc
+
+package assert
+
+import "reflect"
+
+var (
+	boolType = reflect.TypeOf(true)
+)
+
+// seqToSlice checks if x is a sequence, and converts it to a slice of the
+// same element type. Otherwise, x is returned as-is.
+func seqToSlice(x interface{}) interface{} {
```

Comment:
Can't we use type assertion to check it's a sequence
Then use slices.Collect(x) ?

Reply:
no, because:
1. we don’t know the generic type parameter, and
2. i don’t think we can use `iter` and `slices` or generics given the go1.18 version specifier in `go.mod`

Reply:
Indeed, you are not updating Contains signature, and yes the go.mod doesn't help